src/app/pages/api-keys/components/api-key-form-dialog/api-key-form-dialog.component.ts

```diff
@@ -51,26 +42,13 @@ export class ApiKeyFormDialogComponent implements OnInit {
   }
 
   onSubmit(): void {
-    this.loader.open();
     const values = this.form.value;
-    const request$ = this.isNew
-      ? this.ws.call('api_key.create', [{ name: values.name }])
-      : this.ws.call('api_key.update', [this.editingRow.id, values]);
-
-    request$
-      .pipe(untilDestroyed(this))
-      .subscribe((response) => {
-        this.loader.close();
-        this.dialogRef.close(true);
 
-        if (response.key) {
-          this.matDialog.open(KeyCreatedDialogComponent, {
-            data: response.key,
-          });
-        }
-      }, (error) => {
-        this.loader.close();
-        this.errorHandler.handleWsFormError(error, this.form);
-      });
+    if (this.isNew) {
+      this.store$.addApiKey({ name: values.name });
+    } else {
+      this.store$.editApiKey([this.editingRow.id, values]);
```

Comment:
We are missing loading indicators, dialogRef now closes even if operation was unsuccessful and previous error reporting was better in a way that it would show errors in the form.
I believe `addApiKey` and `editApiKey` return observables. You should be able to pipe them in the same way we used to do before and move presentation logic and error handling back here.

Reply:
Yeah, good point. Store `effect` returns `Subscription` type